### ISSUE TITLE
IOS 8 Appview Crash

### ISSUE THREAD

**Issue by Gavin-Harris** [goldfire/howler.js#269]
**How to reproduce:**
Just including the howler library (*howler.core.js*) in an empty html page (*just meta-tags for appview*)
```html
<!DOCTYPE html>
<html lang="en">
	<head>
		<meta name="viewport" content="initial-scale=1, minimum-scale=1, maximum-scale=1, user-scalable=no">
		<meta name="apple-mobile-web-app-status-bar-style" content="black">	
		<meta name="apple-mobile-web-app-capable" content="yes">
		<meta charset="utf-8">
		<title>Howler Test</title>
	</head>
	<body>
		<script src="howler.min.js"></script>
	</body>
</html>
```
Use the Add to Homescreen feature
Launch from srpingboard
Press home button
Immediately tap the app again
Appview just crashes.

**Comment by ashryanbeats** [comment]
I can verify that this happens for the 2.0 branch. Reproducible with this:
http://ashryanbeats.com/apps/guessing-game/index.html

**Comment by TylerFisher** [comment]
Happens on both 2.0, 1.x

**Comment by TylerFisher** [comment]
Did some more investigation. This happens because iOS's UIWebView technically supports Web Audio, but is quite terrible at it and produces this crash across instances of the view. Forcing Howler to use HTML5 on mobile is the graceful fallback.

**Comment by goldfire** [comment]
It wouldn't make sense to just disable Web Audio on mobile since the same problems don't exist in mobile Safari. It could be conceivable to force HTML5 audio in a webview on iOS 8 or lower, but do we have a reliable way of detecting that?

**Comment by kirbyfan64** [comment]
http://stackoverflow.com/a/13549283/2097780?

**Comment by goldfire** [comment]
@kirbyfan64 That gives the iOS version, but what about detecting webview vs Safari?

**Comment by kirbyfan64** [comment]
http://stackoverflow.com/questions/4460205/detect-ipad-iphone-webview-via-javascript?

**Comment by pushpak** [comment]
@goldfire any fix for this ?

**Comment by goldfire** [comment]
This has now been added with 99f27f1f174fe49a1a3572d975a7277c3e723d7d (will be in beta7 later today).

**Comment by TomACPace** [comment]
I'm using a device with iOS 8 on it for testing, and intention to deploy on newer iOS and devices. Found this thread when working backwards in the current library (2.0.3) and found lines 2143-2153.
So far, I am not seeing any crashes; and a (successful!) search on Google found another person commenting on this exact issue from the date this issue was created, and appears to have been resolved in iOS 8.3, April 8 2015. Good that it was in, would have been relevant when the issue was posted.  Anyway it's old and cold. Just a historical retrospective, and note I'm commenting-out line 2151 for my testing.
